app/controllers/application_controller.rb

```diff
@@ -29,6 +29,10 @@ def confirm_admin_user
     redirect_to root_url unless current_user.admin?
   end
 
+  def confirm_admin_user_async
```

Comment:
Not sure if ```async``` is necessary.

Reply:
We already have a `confirm_admin_user` synchronous that redirects to root, sad to say.